### PR TITLE
[TF] Further cleanup of CompilerRuntime and friends

### PR DESCRIPTION
**Pull request by burmako**
I was getting started with TPU support, and this functionality wasn't doing what it was advertising. After a chat with the team, I learned that it was designed to work with GPE, and with GPE gone it doesn't make much sense anymore. Therefore, I decided to go ahead and remove it.

### CODE REVIEW

**Comment by burmako** [comment]
@swift-ci Please test tensorflow

**Comment by burmako** [comment]
@swift-ci Please test tensorflow

File: stdlib/private/TensorFlowUnittest/TensorFlowUnittest.swift [hunk 1]
```diff
@@ -91,9 +90,7 @@ extension TestSuite {
   public func testTPU(_ name: String, _ body: @escaping () -> Void) {
 #if TPU
     test(name + "_TPU") {
```
Comment (rxwei) [inline stdlib/private/TensorFlowUnittest/TensorFlowUnittest.swift]: I think all these test functions are also obsolete actually.

**Review by rxwei — APPROVED** [review]

File: stdlib/private/TensorFlowUnittest/TensorFlowUnittest.swift (hunk 1 above)
Comment (burmako) [reply stdlib/private/TensorFlowUnittest/TensorFlowUnittest.swift]: Further cleanup, here I come!

**Comment by burmako** [comment]
@swift-ci Please test tensorflow

File: stdlib/private/TensorFlowUnittest/TensorFlowUnittest.swift (hunk 1 above)
Comment (burmako) [reply stdlib/private/TensorFlowUnittest/TensorFlowUnittest.swift]: @rxwei I've done some further cleanup. PTAL.

**Comment by burmako** [comment]
@swift-ci Please test tensorflow

**Comment by burmako** [comment]
@swift-ci I'm begging you please test tensorflow

**Review by rxwei — APPROVED** [review]

**Review by bgogul — APPROVED** [review]
Thanks, Eugene!

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Comment by burmako** [comment]
Merging while the macOS job is running, since it is expected that this job will fail anyway.

**Pull request merged**